src/processor.rs

```diff
@@ -27,235 +27,253 @@ impl Processor {
 		    &[&[&b"escrow"[..], &[bump_seed]]], 
 		)?;
 
-		// Close PDA
-		let close_pdas_temp_acc_ix = spl_token::instruction::close_account(
-		    token_program.key,
-		    pda_temp_token_account.key,
-		    initializers_main_account.key,
-		    &pda,
-		    &[&pda]
-		)?;
-		msg!("Calling the token program to close pda's temp account...");
-		invoke_signed(
-		    &close_pdas_temp_acc_ix,
-		    &[
-		        pda_temp_token_account.clone(),
-		        initializers_main_account.clone(),
-		        pda_account.clone(),
-		        token_program.clone(),
-		    ],
-		    &[&[&b"escrow"[..], &[bump_seed]]],
-		)?;
-
-		msg!("Closing the escrow account...");
-		**initializers_main_account.lamports.borrow_mut() = initializers_main_account.lamports()
-			.checked_add(escrow_account.lamports())
-			.ok_or(EscrowError::AmountOverflow)?;
-		**escrow_account.lamports.borrow_mut() = 0;
-		*escrow_account.try_borrow_mut_data()? = &mut [];
-
-		Ok(())
+		Self::close_pda_and_escrow(
+			pda_temp_token_account_info,
+			token_program_account_info,
+			initializers_main_account_info,
+			pda,
+			bump_seed,
+			pda_account_info,
+			escrow_account_info
+		)
 	}
 
 	/// Cancel can be called after init_escrow. If called after exchange, it's already too late
 	/// since exchange is atomic and tokens have been transferred, so there will be no effect.
 	/// Since tokens haven't actually been transferred from initializer main token account to
 	/// initializer temp token account, we don't need to actually transfer any tokens.
 	/// What we need to do is:
-	/// 1) Close escrow info
-	/// 2) Close PDA account
-	fn process_cancel(accounts: &[AccountInfo], amount_expected_to_return: u64, program_id: &Pubkey) -> ProgramResult {
-		let account_info_iter = &mut accounts.iter();
-		let initializer = next_account_info(account_info_iter)?;
+	/// 1) Close PDA account
+	/// 2) Close escrow info
+	fn process_cancel(account_infos: &[AccountInfo], program_id: &Pubkey) -> ProgramResult {
+		let account_info_iter = &mut account_infos.iter();
+		let initializer_info = next_account_info(account_info_iter)?;
 
-		if !initializer.is_signer {
+		if !initializer_info.is_signer {
 			return Err(ProgramError::MissingRequiredSignature);
 		}
 
-		let initializer_token_account = next_account_info(account_info_iter)?;
-		if *initializer_token_account.owner != spl_token::id() {
+		let initializer_token_account_info = next_account_info(account_info_iter)?;
+		if *initializer_token_account_info.owner != spl_token::id() {
 			return Err(ProgramError::IncorrectProgramId);
 		}
 
-		let escrow_account = next_account_info(account_info_iter)?;
-		let escrow_info = Escrow::unpack_unchecked(&escrow_account.try_borrow_data()?)?;
-		if !escrow_info.is_initialized() {
+		let escrow_account_info = next_account_info(account_info_iter)?;
+		let escrow = Escrow::unpack_unchecked(&escrow_account_info.try_borrow_data()?)?;
+		if !escrow.is_initialized() {
 			return Err(ProgramError::UninitializedAccount);
 		}
```

Comment:
this should use `unpack` instead of `unpack_unchecked` (because `unpack` exactly includes the `is_initialized` check)